flang/lib/Optimizer/Builder/Runtime/Reduction.cpp

```diff
@@ -540,9 +606,31 @@ struct ForcedReduceComplex16 {
       auto strTy = fir::ReferenceType::get(mlir::IntegerType::get(ctx, 8));
       auto intTy = mlir::IntegerType::get(ctx, 8 * sizeof(int));
       auto refTy = fir::ReferenceType::get(ty);
+      auto refBoxTy = fir::ReferenceType::get(boxTy);
```

Comment:
Then, I guess, this should not be needed as well :)

Reply:
This is needed because it was actually not tested in the previous patch. I can move this change to the other patch since it's not merged yet.

Reply:
Hmm, I must be missing something.

The runtime API is defined as:
```
void RTDEF(CppReduceComplex16)(std::complex<CppFloat128Type> &result,
    const Descriptor &array,
    ReductionOperation<std::complex<CppFloat128Type>> operation,
    const char *source, int line, int dim, const Descriptor *mask,
    const std::complex<CppFloat128Type> *identity, bool ordered) {
```
So the first argument should be a reference to a complex value, isn't it right?

Reply:
Yes you are right. I messed up the this one with the DIM version. I'll revert it back.